instruqtxblock/instruqtxblock.py

```diff
@@ -89,17 +133,49 @@ def student_view(self, context=None):  # lint-amnesty, pylint: disable=unused-ar
     @XBlock.json_handler
     def completion_handler(self, data, suffix=''):  # lint-amnesty, pylint: disable=unused-argument
         """
-        Handler to trigger completion event
+        Handler to trigger completion and score event
         """
-        save_completion = False
-        try:
-            self.emit_completion(1.0)
-            save_completion = True
-        except BaseException as exp:
-            log.error("Error while marking completion %s", exp)
+        save_track_completion, save_challenge_completion = False, False
+
+        if data['event'] == "track.completed":
```

Comment:
is there a chance `event` won't be in `data`? We can use `.get()` method to be safe.

Reply:
data must have `event` in case it does not we should raise exception`